runtime/src/common/crypto/x25519.rs

```diff
@@ -0,0 +1,125 @@
+//! CBOR serializable X25519 types.
+use anyhow::Result;
+use rand::rngs::OsRng;
+use x25519_dalek;
+use zeroize::Zeroize;
+
+/// The length of an X25519 private key, in bytes.
+pub const PRIVATE_KEY_LENGTH: usize = 32;
+
+/// The length of an X25519 public key, in bytes.
+pub const PUBLIC_KEY_LENGTH: usize = 32;
+
+/// A CBOR serializable Diffie-Hellman X25519 private key.
+#[derive(Clone, Zeroize)]
+#[zeroize(drop)]
+pub struct PrivateKey(pub x25519_dalek::StaticSecret);
+
+impl PrivateKey {
+    /// Generate a new private key.
+    pub fn generate() -> Self {
+        let mut rng = OsRng {};
+
+        PrivateKey(x25519_dalek::StaticSecret::new(&mut rng))
+    }
+
+    /// Compute corresponding public key.
+    pub fn public_key(&self) -> PublicKey {
+        PublicKey(x25519_dalek::PublicKey::from(&self.0))
+    }
+}
+
+impl From<[u8; PRIVATE_KEY_LENGTH]> for PrivateKey {
+    /// Load private key from a byte array.
+    fn from(bytes: [u8; PRIVATE_KEY_LENGTH]) -> PrivateKey {
+        PrivateKey(x25519_dalek::StaticSecret::from(bytes))
+    }
+}
+
+impl Default for PrivateKey {
+    fn default() -> Self {
+        Self::from([0; PRIVATE_KEY_LENGTH])
+    }
+}
+
+impl cbor::Encode for PrivateKey {
+    fn into_cbor_value(self) -> cbor::Value {
+        cbor::to_value(self.0.to_bytes())
+    }
+}
+
+impl cbor::Decode for PrivateKey {
```

Comment:
Add a `try_default` implementation that returns the default value, something like:
```rust
fn try_default() -> Result<Self, DecodeError> {
    Ok(Default::default())
}
```
Same for the public key type.